packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -6,81 +6,62 @@ use postgres_types::ToSql;
-
-        let type_versioned_uri = VersionedUri::new(type_base_uri, type_version as u32);
-        Ok(PersistedEntity::new(
-            entity,
-            entity_id,
-            entity_version,
-            type_versioned_uri,
-            created_by,
-        ))
-    }
-}
-
-#[async_trait]
-impl<C: AsClient> crud::Read<'_, AllLatest, PersistedEntity> for PostgresStore<C> {
-    type Output = Vec<PersistedEntity>;
-
-    async fn get(&self, _: AllLatest) -> Result<Self::Output, QueryError> {
-        let row_stream = self
-            .as_client()
-            .query_raw(
-                r#"
-                SELECT DISTINCT ON(entity_id) properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
-                FROM entities
-                INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
-                ORDER BY entity_id, entities.version DESC;
-                "#,
-                // Requires a concrete type, which implements
-                // `IntoIterator<Item = impl BorrowToSql>`
-                [] as [&(dyn ToSql + Sync); 0],
-            )
-            .await
-            .into_report()
-            .change_context(QueryError)?;
+    async fn read<'query>(
+        &self,
+        query: &Self::Query<'query>,
+    ) -> Result<Vec<PersistedEntity>, QueryError> {
+        let row_stream = match (query.id(), query.version()) {
+            (Some(entity_id), Some(EntityVersion::Latest)) => {
+                self.as_client()
+                    .query_raw(
+                        r#"
+                        SELECT properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+                        FROM entities
+                        INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
+                        WHERE entity_id = $1 AND entities.version = (
+                            SELECT MAX("version")
+                            FROM entities
+                            WHERE entity_id = $1
+                        );
+                        "#,
+                        &[&entity_id],
+                    )
+                    .await
+            }
+            (None, Some(EntityVersion::Latest)) => {
+                self.as_client()
+                    .query_raw(
+                        r#"
+                        SELECT DISTINCT ON(entity_id) properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+                        FROM entities
+                        INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
+                        ORDER BY entity_id, entities.version DESC;
+                        "#,
+                        // Requires a concrete type, which implements
+                        // `IntoIterator<Item = impl BorrowToSql>`
+                        [] as [&(dyn ToSql + Sync); 0],
+                    )
+                    .await
+            }
+            _ => todo!(),
```

Comment:
The plan is to have a general query, that returns basically everything and filter at the Rust side. This is a solution to avoid having a permutation of querys for SQL. This probably can be optimized later, but for the near future this will work just fine.
This PR does not implement the logic because the goal of this PR is to only provide the queries without a breaking change to the REST API. It's not possible to hit this match arm from the REST API.